lib/IRGen/GenCall.cpp

```diff
@@ -1071,11 +1071,12 @@ bool irgen::hasSelfContextParameter(CanSILFunctionType fnType) {
     return metatype->getRepresentation() != MetatypeRepresentation::Thin;
   }
 
-  // Classes and class-bounded archetypes.
+  // Classes and class-bounded archetypes or ObjC existentials.
   // No need to apply this to existentials.
```

Comment:
I guess this comment is slightly inaccurate now, but I'm going to rework this at some point with the objc_method instruction, so it's fine